content/en/docs/Reference/file-based-catalogs.md

```diff
@@ -298,6 +298,10 @@ The `olm.gvk` property [cue][cuelang-spec] schema is:
 
 #### `olm.package.required`
 
+<!-- TODO: deprecate in favor of '#PropertyOLMConstraint: value: package' -->
+
+**Note:** use `#PropertyOLMConstraint: value: package` instead.
```

Comment:
What's the downside of just changing this section of the doc to use this instead of what's documented currently? While we have to deprecate and support `olm.package.required` till it's removed, if we want to encourage new users to use #PropertyOLMConstraint instead, this is the best place to start. If users want information about the old way of mentioning `olm.package.required`, they can be directed to the 0.16.z version of the site.

Reply:
This is Eric's text. We currently have no plan to deprecate `olm.package.required` or `olm.gvk.required`. This is Eric's thinking that they will eventually be deprecated/removed given my generic constraint can do all of them. I do agree with him that until we can have a clear plan of how to deprecate this, we cannot just remove it off main page of our docs. People will visit the main page (latest version) the most and asking them to dig up older version for something that we are currently supporting doesn't make sense. On k8s doc, they maintain text for CRD v1beta1 and v1 all the way until CRD v1beta1 was finally removed in 1.22.

Reply:
In that case how do you feel about adding an example to support the `Note`, so that it's not just a Note that'll be ignored anyway? Also, nit here is to remove the comment from here altogether. An issue for it seems more appropriate for it than a comment in the docs.

Reply:
We decided to remove the notes altogether to avoid confusion. Will add the notes back when we have a concrete plan to deprecate these types.